Sources/NIOCore/ByteBuffer-multi-int.swift

```diff
@@ -0,0 +1,1053 @@
+            v2 = value2.littleEndian
+            v3 = value3.littleEndian
+            v4 = value4.littleEndian
+            v5 = value5.littleEndian
+            v6 = value6.littleEndian
+            v7 = value7.littleEndian
+            v8 = value8.littleEndian
+            v9 = value9.littleEndian
+        }
+
+        let spaceNeeded = Int(MemoryLayout<T1>.size) &+ Int(MemoryLayout<T2>.size) &+ Int(MemoryLayout<T3>.size) &+ Int(MemoryLayout<T4>.size) &+ Int(MemoryLayout<T5>.size) &+ Int(MemoryLayout<T6>.size) &+ Int(MemoryLayout<T7>.size) &+ Int(MemoryLayout<T8>.size) &+ Int(MemoryLayout<T9>.size)
+        return self.writeWithUnsafeMutableBytes(minimumWritableBytes: spaceNeeded) { ptr -> Int in
+            var offset = 0
+            let basePtr = ptr.baseAddress! // safe: pointer is non zero length
+            (basePtr + offset).copyMemory(from: &v1, byteCount: MemoryLayout<T1>.size)
+            offset = offset &+ MemoryLayout<T1>.size
+            (basePtr + offset).copyMemory(from: &v2, byteCount: MemoryLayout<T2>.size)
+            offset = offset &+ MemoryLayout<T2>.size
+            (basePtr + offset).copyMemory(from: &v3, byteCount: MemoryLayout<T3>.size)
+            offset = offset &+ MemoryLayout<T3>.size
+            (basePtr + offset).copyMemory(from: &v4, byteCount: MemoryLayout<T4>.size)
+            offset = offset &+ MemoryLayout<T4>.size
+            (basePtr + offset).copyMemory(from: &v5, byteCount: MemoryLayout<T5>.size)
+            offset = offset &+ MemoryLayout<T5>.size
+            (basePtr + offset).copyMemory(from: &v6, byteCount: MemoryLayout<T6>.size)
+            offset = offset &+ MemoryLayout<T6>.size
+            (basePtr + offset).copyMemory(from: &v7, byteCount: MemoryLayout<T7>.size)
+            offset = offset &+ MemoryLayout<T7>.size
+            (basePtr + offset).copyMemory(from: &v8, byteCount: MemoryLayout<T8>.size)
+            offset = offset &+ MemoryLayout<T8>.size
+            (basePtr + offset).copyMemory(from: &v9, byteCount: MemoryLayout<T9>.size)
+            offset = offset &+ MemoryLayout<T9>.size
+            assert(offset <= ptr.count)
+            return offset
+        }
+    }
+
+    @inlinable
+    public mutating func readMultipleIntegers<T1: FixedWidthInteger, T2: FixedWidthInteger, T3: FixedWidthInteger, T4: FixedWidthInteger, T5: FixedWidthInteger, T6: FixedWidthInteger, T7: FixedWidthInteger, T8: FixedWidthInteger, T9: FixedWidthInteger, T10: FixedWidthInteger>(endianness: Endianness = .big, as: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10).Type = (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10).self) -> (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10)? {
+        let bytesRequired = Int(MemoryLayout<T1>.size) &+ Int(MemoryLayout<T2>.size) &+ Int(MemoryLayout<T3>.size) &+ Int(MemoryLayout<T4>.size) &+ Int(MemoryLayout<T5>.size) &+ Int(MemoryLayout<T6>.size) &+ Int(MemoryLayout<T7>.size) &+ Int(MemoryLayout<T8>.size) &+ Int(MemoryLayout<T9>.size) &+ Int(MemoryLayout<T10>.size)
```

Comment:
The compiler is unhappy with how long it takes to type check this, can we split this up and re-use the sizes too?

Reply:
I think it’ll work with regular +s and given that this is compile-time evaluated we don’t incur overhead for overflow checks

Reply:
I think the compiler can type check it faster if each addition is a separate expressions e.g.:
```
var bytesRequired = 0
bytesRequired &+= Int(MemoryLayout<T1>.size)
bytesRequired &+= Int(MemoryLayout<T2>.size)
bytesRequired &+= Int(MemoryLayout<T3>.size)
...
```
This should result in linear time complexity for type checking because the compiler can find the right overload for just one `&+=` without looking at the other expressions.

Wrapping all calls to `.size` in another `Int` should also not be needed here because [the return type of `MemoryLayout<T*>.size` is already `Int`](https://developer.apple.com/documentation/swift/memorylayout/2432227-size). After that we can simplify it even a bit further:
```
var bytesRequired = 0
bytesRequired &+= MemoryLayout<T1>.size
bytesRequired &+= MemoryLayout<T2>.size
bytesRequired &+= MemoryLayout<T3>.size
...
```